ccmlib/scylla_node.py

```diff
@@ -1571,9 +1580,22 @@ def run_scylla_sstable(self, command, additional_args=None, keyspace=None, dataf
             additional_args = []
 
         scylla_path = common.join_bin(self.get_path(), BIN_DIR, 'scylla')
-        sstables = self._Node__gather_sstables(datafiles, keyspace, column_families)
         ret = {}
 
+        if datafiles is None and keyspace is not None:
+            tag = "sstable-dump-{}".format(uuid.uuid1())
+            kts = ",".join([f"{keyspace}.{column_family}" for column_family in column_families])
+            self.debug(f"run_scylla_sstable(): creating snapshot with tag {tag} to be used for sstable dumping")
+            self.nodetool(f"snapshot -t {tag} {kts}")
+            sstables = []
+            for column_family in column_families:
+                sstables.extend(glob.glob(os.path.join(self.get_path(), 'data', keyspace, f"{column_family}-*/snapshots/{tag}/*-Data.db")))
+        else:
+            sstables = self._Node__gather_sstables(datafiles, keyspace, column_families)
```

Comment:
Out of scope for this PR, possibly another enhancement, but if we expect this function to be called somewhere else, it should not be private.

Reply:
I want to retire this `else:` branch entirely (remove it). But for that I need to survey the usages of `run_scylla_sstable()` and I didn't want to delay this fix. I will do it in a follow-up.

Reply:
To elaborate that a bit: this interface was inherited from the legacy `node.run_sstable2json()`, we wanted a drop-in replacement.
We now know that this is an unsafe interface, because even after this PR, one can still write flaky dtests using this `run_scylla_sstable()`, with certain parameter combinations.
I want to change the API such that it will not be possible to write tests that will fail with disappearing sstables anymore.